packages/headless/src/api/commerce/product-listings/v2/pagination.ts

```diff
@@ -0,0 +1,6 @@
+export interface Pagination {
+  page: number;
```

Comment:
Is it really necessary to specify a page in the response (e.g., returning the same information back from the request)? Same question for `perPage`.

Unless this type is used for both the request _**and**_ response (which I'm not sure I agree with), I'm not convinced these properties need to be here 🙂

Reply:
Because of the way the index works, It's possible that the page that was requested from the API is actually not available. In such a case, the API would presumably return a fallback page instead (the last available page before the requested page, maybe), and we would update the state with this.